File: sdk/c/src/email_and_dns.c
#include "hitechcloud/email_and_dns.h"
#include <stdlib.h>


/* GET /api/infra/dns/rrtypes - Danh mục loại bản ghi DNS kèm số hiệu và ý ngh... */
char *hc_email_and_dns_infra_dns_rrtypes(hc_http_client_t *client) {
    return hc_get(client, "/api/infra/dns/rrtypes", NULL);
}

/* GET /api/infra/dns/rrtypes/2 - Một loại bản ghi DNS */
char *hc_email_and_dns_infra_dns_rrtypes_2(hc_http_client_t *client) {
    return hc_get(client, "/api/infra/dns/rrtypes/2", NULL);
}

/* GET /api/infra/email/auth/methods - Phương thức xác thực thư và tên kết quả xuất h... */
char *hc_email_and_dns_infra_email_auth_methods(hc_http_client_t *client) {
    return hc_get(client, "/api/infra/email/auth/methods", NULL);
}

/* GET /api/infra/email/status/codes - Danh mục mã trạng thái SMTP mở rộng */
char *hc_email_and_dns_infra_email_status_codes(hc_http_client_t *client) {
    return hc_get(client, "/api/infra/email/status/codes", NULL);
}

/* GET /api/infra/email/status/codes/2 - Giải nghĩa một mã trạng thái SMTP mở rộng như... */
char *hc_email_and_dns_infra_email_status_codes_2(hc_http_client_t *client) {
    return hc_get(client, "/api/infra/email/status/codes/2", NULL);
}

/* GET /api/infra/email/tags - Ý nghĩa từng thẻ trong bản ghi DKIM và DMARC */
char *hc_email_and_dns_infra_email_tags(hc_http_client_t *client) {
    return hc_get(client, "/api/infra/email/tags", NULL);
}

/* GET /api/tools/dns/email/blacklist - Đối chiếu tên miền hoặc IP với các danh sách đ... */
char *hc_email_and_dns_tools_dns_email_blacklist(hc_http_client_t *client) {
    return hc_get(client, "/api/tools/dns/email/blacklist", NULL);
}

/* POST /api/tools/dns/email/check - Kiểm tra một lượt cả SPF */
char *hc_email_and_dns_tools_dns_email_check(hc_http_client_t *client) {
    return hc_post(client, "/api/tools/dns/email/check", NULL);
}

/* POST /api/tools/dns/email/header - Phân tích header thư: đường đi */
char *hc_email_and_dns_tools_dns_email_header(hc_http_client_t *client) {
    return hc_post(client, "/api/tools/dns/email/header", NULL);
}

/* POST /api/tools/dns/lookup - Truy vấn bản ghi DNS trực tiếp */
char *hc_email_and_dns_tools_dns_lookup(hc_http_client_t *client) {
    return hc_post(client, "/api/tools/dns/lookup", NULL);
}

/* POST /api/tools/dns/propagation - Kiểm tra bản ghi DNS đã lan truyền tới các máy... */
char *hc_email_and_dns_tools_dns_propagation(hc_http_client_t *client) {
    return hc_post(client, "/api/tools/dns/propagation", NULL);
}

/* POST /api/tools/dns/smtp/test - Thử kết nối SMTP tới máy chủ thư để xem có nhậ... */
char *hc_email_and_dns_tools_dns_smtp_test(hc_http_client_t *client) {
    return hc_post(client, "/api/tools/dns/smtp/test", NULL);
}

/* POST /api/tools/email/blacklist/checker - Kiểm tra tên miền hoặc IP có nằm trong danh sá... */
char *hc_email_and_dns_tools_email_blacklist_checker(hc_http_client_t *client) {
    return hc_post(client, "/api/tools/email/blacklist/checker", NULL);
}

/* GET /api/tools/email/dkim/checker - Kiểm tra bản ghi DKIM */
char *hc_email_and_dns_tools_email_dkim_checker(hc_http_client_t *client) {
    return hc_get(client, "/api/tools/email/dkim/checker", NULL);
}

/* GET /api/tools/email/dmarc/checker - Kiểm tra chính sách DMARC */
char *hc_email_and_dns_tools_email_dmarc_checker(hc_http_client_t *client) {
    return hc_get(client, "/api/tools/email/dmarc/checker", NULL);
}

/* GET /api/tools/email/mx/checker - Danh sách máy chủ nhận thư của tên miền */
char *hc_email_and_dns_tools_email_mx_checker(hc_http_client_t *client) {
    return hc_get(client, "/api/tools/email/mx/checker", NULL);
}

/* GET /api/tools/email/spf/checker - Kiểm tra bản ghi SPF của tên miền */
char *hc_email_and_dns_tools_email_spf_checker(hc_http_client_t *client) {
    return hc_get(client, "/api/tools/email/spf/checker", NULL);
}

File: sdk/r/R/email_and_dns.R
#' Email & DNS
#' @param http HttpClient object
#' @return Resource object
#' @export
EmailAndDnsResource <- function(http) {
  env <- new.env(parent = emptyenv())
  env$http <- http
  class(env) <- "EmailAndDnsResource"
  env
}

#' GET /api/infra/dns/rrtypes - Danh mục loại bản ghi DNS kèm số hiệu và ý ngh...
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
infraDnsRrtypes <- function(self, ...) {
  self$http$get("/api/infra/dns/rrtypes", list(...))
}

#' GET /api/infra/dns/rrtypes/2 - Một loại bản ghi DNS
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
infraDnsRrtypes2 <- function(self, ...) {
  self$http$get("/api/infra/dns/rrtypes/2", list(...))
}

#' GET /api/infra/email/auth/methods - Phương thức xác thực thư và tên kết quả xuất h...
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
infraEmailAuthMethods <- function(self, ...) {
  self$http$get("/api/infra/email/auth/methods", list(...))
}

#' GET /api/infra/email/status/codes - Danh mục mã trạng thái SMTP mở rộng
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
infraEmailStatusCodes <- function(self, ...) {
  self$http$get("/api/infra/email/status/codes", list(...))
}

#' GET /api/infra/email/status/codes/2 - Giải nghĩa một mã trạng thái SMTP mở rộng như...
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
infraEmailStatusCodes2 <- function(self, ...) {
  self$http$get("/api/infra/email/status/codes/2", list(...))
}

#' GET /api/infra/email/tags - Ý nghĩa từng thẻ trong bản ghi DKIM và DMARC
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
infraEmailTags <- function(self, ...) {
  self$http$get("/api/infra/email/tags", list(...))
}

#' GET /api/tools/dns/email/blacklist - Đối chiếu tên miền hoặc IP với các danh sách đ...
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
toolsDnsEmailBlacklist <- function(self, ...) {
  self$http$get("/api/tools/dns/email/blacklist", list(...))
}

#' POST /api/tools/dns/email/check - Kiểm tra một lượt cả SPF
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
toolsDnsEmailCheck <- function(self, ...) {
  self$http$post("/api/tools/dns/email/check", list(...))
}

#' POST /api/tools/dns/email/header - Phân tích header thư: đường đi
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
toolsDnsEmailHeader <- function(self, ...) {
  self$http$post("/api/tools/dns/email/header", list(...))
}

#' POST /api/tools/dns/lookup - Truy vấn bản ghi DNS trực tiếp
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
toolsDnsLookup <- function(self, ...) {
  self$http$post("/api/tools/dns/lookup", list(...))
}

#' POST /api/tools/dns/propagation - Kiểm tra bản ghi DNS đã lan truyền tới các máy...
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
toolsDnsPropagation <- function(self, ...) {
  self$http$post("/api/tools/dns/propagation", list(...))
}

#' POST /api/tools/dns/smtp/test - Thử kết nối SMTP tới máy chủ thư để xem có nhậ...
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
toolsDnsSmtpTest <- function(self, ...) {
  self$http$post("/api/tools/dns/smtp/test", list(...))
}

#' POST /api/tools/email/blacklist/checker - Kiểm tra tên miền hoặc IP có nằm trong danh sá...
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
toolsEmailBlacklistChecker <- function(self, ...) {
  self$http$post("/api/tools/email/blacklist/checker", list(...))
}

#' GET /api/tools/email/dkim/checker - Kiểm tra bản ghi DKIM
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
toolsEmailDkimChecker <- function(self, ...) {
  self$http$get("/api/tools/email/dkim/checker", list(...))
}

#' GET /api/tools/email/dmarc/checker - Kiểm tra chính sách DMARC
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
toolsEmailDmarcChecker <- function(self, ...) {
  self$http$get("/api/tools/email/dmarc/checker", list(...))
}

#' GET /api/tools/email/mx/checker - Danh sách máy chủ nhận thư của tên miền
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
toolsEmailMxChecker <- function(self, ...) {
  self$http$get("/api/tools/email/mx/checker", list(...))
}

#' GET /api/tools/email/spf/checker - Kiểm tra bản ghi SPF của tên miền
#' @param self Resource object
#' @param ... Additional parameters
#' @return API response
toolsEmailSpfChecker <- function(self, ...) {
  self$http$get("/api/tools/email/spf/checker", list(...))
}

